app/Http/Controllers/BaseHomeController.php

```diff
@@ -203,4 +204,87 @@ public function updateLatestVersion(Request $request)
             return $result;
         }
     }
+    
+    public function updateLicenseCode(Request $request)
+    {
+        try {
```

Comment:
I have tried refactoring it. You can come up with your version
```
 try {
            $licCode = $request->input('licenseCode');//The license code already existing for older client
            $lastFour = $this->getLastFourDigistsOfLicenseCode($request->input('product'));
          
            $existingLicense = Order::all()->select("id", "client", "product")
                ->filter(function ($order) use ($licCode) {
                    return $order->serial_key == $licCode
                })->first();

            if (existingLicense) {//If the license code that is sent in the request exists in billing 
                $cont = new \App\Http\Controllers\License\LicenseController();
                $cont->updateInstalledDomain($licCode,$existingLicense->product);//Delete the installation first for the current license before updating license so that no Faveo installation exists on the user domain/IP path 
                
                $serial_key = substr($licCode, 0,12).$lastFour;//The new License Code
                //Create new license in license manager with the new license code which has no. of agents in the last 4 digits.
                $cont->createNewLicene(
                    $existingLicense->id, 
                    $existingLicense->product, 
                    $existingLicense->product->client, 
                    $this->getLicenseExpiryDate($existingLicense), 
                    $this->getUpdatesExpiryDate($existingLicense), 
                    $this->getSupportExpiryDate($existingLicense), 
                    $serial_key
                );
                
                //Update the old license code with new one in billing. 
                $existingLicense->serial_key = \Crypt::encrypt(substr($licCode, 0,12).$lastFour);
                $existingLicense->save();
                
                //send the newly updated license code in response
                return response()->json(['status'=>'success', 'updatedLicenseCode'=>$existingLicense->serial_key]);
            }
        } catch (\Exception $ex) {
            return response()->json(['status'=>'fails', 'error' => $ex->getMessage()]);
        }
```